model_compression_toolkit/core/common/quantization/candidate_node_quantization_config.py

```diff
@@ -32,34 +29,22 @@ class CandidateNodeQuantizationConfig(BaseNodeNodeQuantizationConfig):
     """
     Class for representing candidate node configuration, which includes weights and activation configuration combined.
     """
-    def __init__(self,
-                 qc: QuantizationConfig,
-                 op_cfg: OpQuantizationConfig,
-                 activation_quantization_fn: Callable,
-                 activation_quantization_params_fn: Callable,
-                 weights_quantization_fn: Callable,
-                 weights_quantization_params_fn: Callable,
-                 weights_channels_axis: int
-                 ):
-        """
-
-        Args:
-            qc: QuantizationConfig to create the node's config from.
-            op_cfg: OpQuantizationConfig of the node with quantizers types to use when creating node quantization configuration.
-            activation_quantization_fn: Function to use when quantizing the node's activations.
-            activation_quantization_params_fn: Function to use when computing the threshold for quantizing a node's activations.
-            weights_quantization_fn: Function to use when quantizing the node's weights.
-            weights_quantization_params_fn:  Function to use when computing the threshold for quantizing a node's weights.
-            weights_channels_axis: Axis to quantize a node's kernel when quantizing per-channel.
-        """
-
-        self.activation_quantization_cfg = NodeActivationQuantizationConfig(qc,
-                                                                            op_cfg,
-                                                                            activation_quantization_fn,
-                                                                            activation_quantization_params_fn)
 
-        self.weights_quantization_cfg = NodeWeightsQuantizationConfig(qc,
-                                                                      op_cfg,
-                                                                      weights_quantization_fn,
-                                                                      weights_quantization_params_fn,
-                                                                      weights_channels_axis)
+    def __init__(self, **kwargs):
```

Comment:
I'm ok with this, but I think we can change init to receive quantization configurations and add a method that creates quantization configurations from "old params" and init it.

Reply:
I think that this way is more elegant, but if you want I can change this.